Lib/tarfile.py

```diff
@@ -1604,17 +1604,20 @@ def open(cls, name=None, mode="r", fileobj=None, bufsize=RECORDSIZE, **kwargs):
             # Find out which *open() is appropriate for opening the file.
             def not_compressed(comptype):
                 return cls.OPEN_METH[comptype] == 'taropen'
+            errors = []
             for comptype in sorted(cls.OPEN_METH, key=not_compressed):
                 func = getattr(cls, cls.OPEN_METH[comptype])
                 if fileobj is not None:
                     saved_pos = fileobj.tell()
                 try:
                     return func(name, "r", fileobj, **kwargs)
-                except (ReadError, CompressionError):
+                except (ReadError, CompressionError) as e:
```

Comment:
I am ok with this, but this is going to create a hell of a lot of cycles, so I prefer to manually delete the list with all the exceptions once `error_msgs` is done to alleviate this a bit.

Reply:
I actually thought about the reference cycle problem -- that's why these are strings and not exception objects -- I don't think there's actually any cycles added by this PR ?  unless you're hinting at something I don't know about!

Reply:
> I don't think there's actually any cycles added by this PR ?

Oh, my bad, I was reading this as keeping the errors in the list instead of the string representation. Maybe renaming `errors` to `error_messages`? If you prefer not to do that is fine, I think we can go as it is as well.

Reply:
ah yeah that would be better -- but then what to call the joined variable 🤔 (to avoid having a variable change type)

Reply:
> the joined variable

`final_error_msg `?

Reply:
Notice that even if it contains the summary of multiple error messages is a single error message at the end.

Maybe you can call it `summary_error_msg`?

Reply:
went with `error_msgs` and `error_msgs_summary` -- thanks for the suggestion!